typescript/submissions.d.ts

```diff
@@ -210,3 +215,313 @@ export type FormStoreDefinition = {
+    transactionToken: string
+    merchantCode: string
+    orderNumber: string
+    chargeTypeId: 1 | 2 | 3 | 4 | 5 | 6 | 7
+    creditCardTypeId: 0 | 1 | 2 | 3 | 4 | 5 | 6
+    paymentTypeId: 1 | 2
+    amount: number
+    lastFour: string
+    expMonth: number
+    expYear: number
+    resultCode: number
+    errorCode: CPPayTransactionErrorCodes
+    errorMessage: string
+    savedPaymentMethodToken: string
+    customerReceipt: string
+    merchantReceipt: string
+    initialChargeStatusId: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8
+    currentChargeStatusId: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8
+    currentChargeStatusDtm: Date
+    createdAt: Date
+    updatedAt: Date
+    deletedAt: Date
+    customerSignature: string
+    customerSignatureFormat: string
+    isSuccessSavedPaymentMethod: boolean
+  }
+}
+
+export type CPPayTransactionErrorCodes =
+  | 0
+  | 400
+  | 401
+  | 404
+  | 503
+  | 1001
+  | 1002
+  | 1003
+  | 1004
+  | 1005
+  | 1006
+  | 1007
+  | 1008
+  | 1009
+  | 1010
+  | 1012
+  | 1013
+  | 2001
+  | 2002
+  | 2003
+  | 2004
+  | 2005
+  | 2006
+  | 2007
+  | 2008
+  | 2009
+  | 2010
+  | 2011
+  | 2012
+  | 2013
+  | 2014
+  | 2015
+  | 2016
+  | 2017
+  | 2018
+  | 2019
+  | 2020
+  | 2021
+  | 2022
+  | 2023
+  | 2024
+  | 2025
+  | 2026
+  | 2027
+  | 2028
+  | 2029
+  | 2030
+  | 2031
+  | 2032
+  | 2033
+  | 2034
+  | 2035
+  | 2036
+  | 2050
+  | 2100
+  | 2101
+  | 2102
+  | 3001
+  | 3002
+  | 3003
+  | 3004
+  | 3005
+  | 3006
+  | 3007
+  | 3008
+  | 3009
+  | 3010
+  | 3011
+  | 3012
+  | 3013
+  | 3014
+  | 3015
+  | 3016
+  | 3017
+  | 3018
+  | 3019
+  | 3020
+  | 3021
+  | 3022
+  | 3023
+  | 3024
+  | 3025
+  | 3026
+  | 3027
+  | 3028
+  | 4001
+  | 4002
+  | 5001
+  | 5002
+  | 5003
+  | 5004
+  | 5005
+  | 5006
+  | 5007
+  | 5008
+  | 5009
+  | 5010
+  | 5011
+  | 5012
+  | 6000
+  | 6001
+  | 6002
+  | 6003
+  | 6004
+  | 6005
+  | 6006
+  | 6007
+  | 6008
+  | 6009
+  | 6010
+  | 6011
+  | 6012
+  | 6013
+  | 6014
+  | 6015
+  | 6016
+  | 6017
+  | 6018
+  | 6019
+  | 6020
+  | 7001
+  | 7002
+  | 7004
+  | 7005
+
+export type NewFormSubmissionPayment = {
+  submissionId: string
+  formId: number
+  status: 'PENDING' | 'SUCCEEDED' | 'FAILED'
+}
+
+export type WestpacQuickWebPayment = NewFormSubmissionPayment & {
+  type: WestpacQuickWebSubmissionEvent['type']
+  createdAt: string
+  updatedAt: string
+  paymentTransaction?: {
+    sourceCode?: string
+    receiptNumber?: string
+    communityCode?: string
+    supplierBusinessCode?: string
+    paymentReference: string
+    customerReferenceNumber?: string
+    paymentAmount?: number
+    surchargeAmount?: number
+    cardScheme?: string
+    settlementDate?: string
+    createdDateTime?: string
+    responseCode?: string
+    responseDescription?: string
+    successFlag: boolean
+  }
+}
+
+export type BPOINTPayment = NewFormSubmissionPayment & {
+  type: BPOINTSubmissionEvent['type']
+  createdAt: string
+  updatedAt: string
+  paymentTransaction?: {
```

Comment:
Might make the API easier to implement if this property exists on `NewFormSubmissionPayment` instead of `FormSubmissionPayment`